File: src_original/extras-attica/liblicense.cpp
#define WORKAROUND_INNER_CLASS_DEFINITION_Attica__License
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QUrl>
#include <license.h>
#include "liblicense.h"
#include "liblicense.hxx"

Attica__License* Attica__License_new() {
    return new Attica::License();
}

Attica__License* Attica__License_new2(const Attica__License* other) {
    return new Attica::License(*other);
}

void Attica__License_OperatorAssign(Attica__License* self, const Attica__License* other) {
    self->operator=(*other);
}

unsigned int Attica__License_Id(const Attica__License* self) {
    return static_cast<unsigned int>(self->id());
}

void Attica__License_SetId(Attica__License* self, unsigned int id) {
    self->setId(static_cast<uint>(id));
}

libqt_string Attica__License_Name(const Attica__License* self) {
    QString _ret = self->name();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

void Attica__License_SetName(Attica__License* self, const libqt_string name) {
    QString name_QString = QString::fromUtf8(name.data, name.len);
    self->setName(name_QString);
}

QUrl* Attica__License_Url(const Attica__License* self) {
    return new QUrl(self->url());
}

void Attica__License_SetUrl(Attica__License* self, const QUrl* url) {
    self->setUrl(*url);
}

void Attica__License_Delete(Attica__License* self) {
    delete self;
}

File: include_original/extras-attica/liblicense.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const std = @import("std");

/// https://api.kde.org/attica-license.html
pub const attica__license = struct {
    /// New constructs a new Attica::License object.
    ///
    ///
    pub fn New() QtC.Attica__License {
        return qtc.Attica__License_new();
    }

    /// New2 constructs a new Attica::License object.
    ///
    /// ``` other: QtC.Attica__License ```
    pub fn New2(other: ?*anyopaque) QtC.Attica__License {
        return qtc.Attica__License_new2(@ptrCast(other));
    }

    /// [Qt documentation](https://api.kde.org/attica-license.html#operator-eq)
    ///
    /// ``` self: QtC.Attica__License, other: QtC.Attica__License ```
    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.Attica__License_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://api.kde.org/attica-license.html#id)
    ///
    /// ``` self: QtC.Attica__License ```
    pub fn Id(self: ?*anyopaque) u32 {
        return qtc.Attica__License_Id(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/attica-license.html#setId)
    ///
    /// ``` self: QtC.Attica__License, id: u32 ```
    pub fn SetId(self: ?*anyopaque, id: u32) void {
        qtc.Attica__License_SetId(@ptrCast(self), @intCast(id));
    }

    /// [Qt documentation](https://api.kde.org/attica-license.html#name)
    ///
    /// ``` self: QtC.Attica__License, allocator: std.mem.Allocator ```
    pub fn Name(self: ?*anyopaque, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.Attica__License_Name(@ptrCast(self));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("attica::license.Name: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// [Qt documentation](https://api.kde.org/attica-license.html#setName)
    ///
    /// ``` self: QtC.Attica__License, name: []const u8 ```
    pub fn SetName(self: ?*anyopaque, name: []const u8) void {
        const name_str = qtc.libqt_string{
            .len = name.len,
            .data = name.ptr,
        };
        qtc.Attica__License_SetName(@ptrCast(self), name_str);
    }

    /// [Qt documentation](https://api.kde.org/attica-license.html#url)
    ///
    /// ``` self: QtC.Attica__License ```
    pub fn Url(self: ?*anyopaque) QtC.QUrl {
        return qtc.Attica__License_Url(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/attica-license.html#setUrl)
    ///
    /// ``` self: QtC.Attica__License, url: QtC.QUrl ```
    pub fn SetUrl(self: ?*anyopaque, url: ?*anyopaque) void {
        qtc.Attica__License_SetUrl(@ptrCast(self), @ptrCast(url));
    }

    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.Attica__License ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.Attica__License_Delete(@ptrCast(self));
    }
};
